packages/web/src/js/FileUploader.ts

```diff
@@ -0,0 +1,277 @@
+    this.fileQueueLimit = element.dataset.fileQueueLimit
+      ? Number(element.dataset.fileQueueLimit)
+      : DEFAULT_FILE_QUEUE_LIMIT;
+    this.inputName = this.inputElement?.name || 'attachment';
+    this.isMultiple = this.inputElement?.multiple;
+    this.isDragging = false;
+    this.fileQueue = new Map();
+
+    this.init();
+  }
+
+  static get NAME() {
+    return NAME;
+  }
+
+  get getFileQueue() {
+    return this.fileQueue;
+  }
+
+  public clearFileQueue(): void {
+    this.fileQueue.clear();
+
+    while (this.listElement?.firstChild) {
+      this.listElement.removeChild(this.listElement.firstChild);
+    }
+  }
+
+  static getUpdatedFileName(name: string): string {
+    return `file__${name.replace(/\./g, '_')}`;
+  }
+
+  dragReset() {
+    this.isDragging = false;
+    this.wrapper?.classList.remove(IS_DRAGGING_CLASS_NAME);
+  }
+
+  checkAllowedFileSize(file: File) {
+    if (file.size > this.fileSizeLimit) {
+      throw new Error('The file size limit has been exceeded ⚠️');
+    }
+  }
+
+  checkFileQueueDuplicity(file: File) {
+    if (this.fileQueue.has(FileUploader.getUpdatedFileName(file.name))) {
+      throw new Error('This file already exists in the queue ⚠️');
+    }
+  }
+
+  checkQueueLimit() {
+    if (this.fileQueue.size >= this.fileQueueLimit) {
+      throw new Error('You have exceeded the number of files allowed in the queue ⚠');
+    }
+  }
+
+  createAttachmentElement() {
+    const attachmentInputElement = document.createElement('input');
+    attachmentInputElement.setAttribute('type', 'file');
+    attachmentInputElement.setAttribute('name', `${this.inputName}[]`);
+    attachmentInputElement.setAttribute('hidden', '');
+
+    return attachmentInputElement;
+  }
+
+  getAttachmentElement(file: File, id: string): HTMLElement | null {
+    const snippetTemplate = SelectorEngine.findOne(TEMPLATE_ELEMENT_SELECTOR, this.element) as HTMLTemplateElement;
+    const snippet = snippetTemplate?.content.cloneNode(true) as HTMLElement;
+
+    if (!snippet) {
+      return null;
+    }
+
+    const dataContainer = new DataTransfer();
+    dataContainer.items.add(file);
+
+    const attachmentInputElement = this.createAttachmentElement();
+    attachmentInputElement.files = dataContainer.files;
+
+    const item = snippet.querySelector(`[${TEMPLATE_ELEMENT_SLOT_NAME}="item"]`);
+    const itemName = snippet.querySelector(`[${TEMPLATE_ELEMENT_SLOT_NAME}="name"]`);
+    const itemButton = snippet.querySelector(`[${TEMPLATE_ELEMENT_SLOT_NAME}="button"]`);
+
+    item!.appendChild(attachmentInputElement);
+    item!.setAttribute('id', id);
+    itemName!.innerHTML = file.name;
+    itemButton!.setAttribute(DATA_DISMISS_ATTRIBUTE, id);
+    item!.removeAttribute(TEMPLATE_ELEMENT_SLOT_NAME);
+    itemName!.removeAttribute(TEMPLATE_ELEMENT_SLOT_NAME);
+    itemButton!.removeAttribute(TEMPLATE_ELEMENT_SLOT_NAME);
+
+    return snippet;
+  }
+
+  createAttachmentEvents(id: string) {
```

Comment:
Maybe `addAttachmentEventListeners` because we have `addEventListeners`?

Reply:
But we also have `createAttachmentElement`